### PR TITLE
fix(www): fallback to github & unpkg to get README.md

### LINKED ISSUE #11129: [www] Empty pages on plugin pages
Some plugin pages have empty pages:
https://www.gatsbyjs.org/packages/gatsby-source-sanity/?=sanity

I've did some digging and found out that the api we're calling is responding with an empty readme property. So it's not an issue on our end but maybe we want to setup or own endpoint so we're not dependant on algolia.

The plugin we're using is [gatsby-source-npm-package-search](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-npm-package-search) which uses [algolia/npm-search](https://github.com/algolia/npm-search)
<details>
  <summary>API output for gatsby-source-sanity</summary>

```js
{
  "name": "gatsby-source-sanity",
  "downloadsLast30Days": 985,
  "downloadsRatio": 0.00000340270253936103,
  "humanDownloadsLast30Days": "985",
  "popular": false,
  "version": "0.5.2",
  "versions": {
    "0.1.0": "2018-07-29T15:02:07.472Z",
    "0.1.1": "2018-07-29T15:16:36.366Z",
    "0.1.2": "2018-07-29T15:22:40.294Z",
    "0.1.3": "2018-07-29T15:37:06.989Z",
    "0.2.0": "2018-08-01T20:26:49.582Z",
    "0.2.1": "2018-08-01T23:44:35.157Z",
    "0.2.2": "2018-08-02T00:18:57.848Z",
    "0.3.0": "2018-08-03T14:15:07.977Z",
    "0.3.1": "2018-08-03T19:48:16.057Z",
    "0.3.2": "2018-08-03T21:30:30.058Z",
    "0.3.3": "2018-08-04T14:32:47.493Z",
    "0.3.4": "2018-08-10T22:53:03.804Z",
    "0.4.0": "2018-08-13T17:19:19.901Z",
    "0.4.1": "2018-08-22T01:13:38.415Z",
    "0.5.0": "2018-10-12T13:33:49.457Z",
    "0.5.1": "2018-10-12T13:38:16.219Z",
    "0.9.10-v1.0": "2019-01-02T14:46:44.971Z",
    "0.9.10-next.0": "2019-01-06T22:00:22.368Z",
    "0.9.10-next.1": "2019-01-07T20:41:09.409Z",
    "0.9.10-next.2": "2019-01-07T21:01:42.487Z",
    "0.9.10-next.3": "2019-01-08T10:43:59.328Z",
    "0.9.10-next.4": "2019-01-08T13:13:08.858Z",
    "0.9.10-next.5": "2019-01-11T10:00:43.749Z",
    "0.9.10-next.6": "2019-01-11T12:31:03.702Z",
    "0.9.10-next.7": "2019-01-12T00:33:48.630Z",
    "0.5.2": "2019-01-14T08:59:14.348Z",
    "0.9.10-next.8": "2019-01-14T18:32:49.425Z",
    "0.9.10-next.9": "2019-01-14T22:15:15.077Z",
    "0.9.10-next.10": "2019-01-15T09:27:54.183Z",
    "0.9.10-next.11": "2019-01-15T12:17:31.262Z"
  },
  "tags": {
    "latest": "0.5.2",
    "typed": "0.5.0",
    "next": "0.9.10-next.11"
  },
  "description": "Gatsby source plugin for building websites using Sanity.io as a backend.",
  "dependencies": {
    "@sanity/client": "^0.132.5",
    "@sanity/image-url": "^0.132.5",
    "gatsby-source-filesystem": "^2.0.0"
  },
  "devDependencies": {
    "@types/node": "^10.11.7",
    "cross-env": "^5.1.3",
    "husky": "^1.1.2",
    "prettier": "^1.14.3",
    "pretty-quick": "^1.7.0",
    "tslint": "^5.11.0",
    "tslint-config-prettier": "^1.15.0",
    "typescript": "^3.1.3"
  },
  "repository": {
    "url": "https://github.com/hcavalieri/gatsby-source-sanity",
    "project": "gatsby-source-sanity",
    "user": "hcavalieri",
    "host": "github.com",
    "path": "",
    "head": "dcb03b586165bb630ac924df865eacb56d7631ba",
    "branch": "dcb03b586165bb630ac924df865eacb56d7631ba"
  },
  "githubRepo": {
    "user": "hcavalieri",
    "project": "gatsby-source-sanity",
    "path": "",
    "head": "dcb03b586165bb630ac924df865eacb56d7631ba"
  },
  "gitHead": "dcb03b586165bb630ac924df865eacb56d7631ba",
  "readme": "",
  "owner": {
    "name": "hcavalieri",
    "avatar": "https://github.com/hcavalieri.png",
    "link": "https://github.com/hcavalieri"
  },
  "deprecated": false,
  "homepage": null,
  "license": "MIT",
  "keywords": [
    "gatsby",
    "gatsby-plugin",
    "gatsby-source-plugin"
  ],
  "computedKeywords": [],
  "computedMetadata": {},
  "created": 1532876527404,
  "modified": 1547554653826,
  "lastPublisher": {
    "name": "rexxars",
    "email": "espen@hovlandsdal.com",
    "avatar": "https://gravatar.com/avatar/3e1447c9a6e5aed088af8ebc4297ffdb",
    "link": "https://www.npmjs.com/~rexxars"
  },
  "owners": [
    {
      "email": "bjoerge@gmail.com",
      "name": "bjoerge",
      "avatar": "https://gravatar.com/avatar/b60ee201135f11aabeb1c2d2eccfa728",
      "link": "https://www.npmjs.com/~bjoerge"
    },
    {
      "email": "henrique@kaordica.com.br",
      "name": "hcavalieri",
      "avatar": "https://gravatar.com/avatar/0273bdb35d9bda75bc256e75f1f9d65e",
      "link": "https://www.npmjs.com/~hcavalieri"
    },
    {
      "email": "studio@mariuslundgard.com",
      "name": "mariuslundgard",
      "avatar": "https://gravatar.com/avatar/52519966a7085e1720797218e9b65114",
      "link": "https://www.npmjs.com/~mariuslundgard"
    },
    {
      "email": "espen@hovlandsdal.com",
      "name": "rexxars",
      "avatar": "https://gravatar.com/avatar/3e1447c9a6e5aed088af8ebc4297ffdb",
      "link": "https://www.npmjs.com/~rexxars"
    },
    {
      "email": "developers@sanity.io",
      "name": "sanity-io",
      "avatar": "https://gravatar.com/avatar/cbeeb446d0c1169743bf453664abba2b",
      "link": "https://www.npmjs.com/~sanity-io"
    }
  ],
  "lastCrawl": "2019-01-15T12:17:34.742Z",
  "_searchInternal": {
    "concatenatedName": "gatsbysourcesanity",
    "alternativeNames": [
      "gatsbysourcesanity",
      "gatsby source sanity",
      "gatsby-source-sanity"
    ],
    "downloadsMagnitude": 3,
    "jsDelivrPopularity": 0
  },
  "dependents": 0,
  "humanDependents": "0",
  "changelogFilename": null,
  "jsDelivrHits": 0,
  "objectID": "gatsby-source-sanity"
}
```
</detail>

### LINKED ISSUE #16318: Gatsby Plugins Library broken with gatsby themes
## Description

Clicking on any gatsby plugin in the plugins library causes 404

### Steps to reproduce

go here https://www.gatsbyjs.org/plugins/

and click on any gatsby theme

### Expected result

it should work

### Actual result

image

### PR DESCRIPTION
**Pull request by wardpeet**
## Description
Fixes empty readme files.

This is the debug log when falling back to github & unpkg
```
readme created from github for gatsby-plugin-metomic
readme created from github for gatsby-transformer-sqip
readme created from github for gatsby-source-prismic
readme created from github for gatsby-plugin-sharp
readme created from github for gatsby-source-graphql
readme created from github for @hakrsingh/gatsby-theme-chaos
readme created from github for gatsby-plugin-algolia-docusearch
readme created from github for @monastic.panic/gatsby-transformer-react-docgen
readme created from github for gatsby-transformer-video
readme created from github for gatsby-source-prismic-graphql
readme created from github for @wonism/gatsby-source-wordpress
readme created from unpkg for gatsby-plugin-freshdesk
readme created from unpkg for gatsby-plugin-mdx-prismjs
readme created from unpkg for @nteract/gatsby-transformer-ipynb
readme created from unpkg for gatsby-plugin-google-adsense
readme created from unpkg for gatsby-plugin-eslint-v2
readme created from unpkg for gatsby-plugin-react-next
readme created from unpkg for @wardpeet/gatsby-plugin-static-site
```

It's based on [@Haroenv comment](https://github.com/gatsbyjs/gatsby/issues/11129#issuecomment-466474445)

## Related Issues
Fixes #16318
Fixes #11129

### CODE REVIEW

File: packages/gatsby-source-npm-package-search/src/gatsby-node.js [hunk 1]
```diff
@@ -28,8 +48,29 @@ exports.sourceNodes = async (
     hitsPerPage: 1000,
   })
 
-  hits.forEach(hit => {
+  await hits.map(async hit => {
```
Comment (m-allanson) [inline packages/gatsby-source-npm-package-search/src/gatsby-node.js]: Should this be `await Promise.all(hits.map(/* foo bar baz */))`?

File: packages/gatsby-source-npm-package-search/src/gatsby-node.js [hunk 2]
```diff
@@ -28,8 +48,29 @@ exports.sourceNodes = async (
...
+        )).body
+      } catch (err) {
+        // carry-on
+      }
+    }
+
+    if (!hit.readme) {
+      try {
+        hit.readme = (await got.get(
+          url.resolve(`https://unpkg.com/`, `/${hit.objectID}/README.md`)
```
Comment (m-allanson) [inline packages/gatsby-source-npm-package-search/src/gatsby-node.js]: What's the reason for using an unpkg url instead of GitHub?

Comment (sidharthachatterjee) [reply packages/gatsby-source-npm-package-search/src/gatsby-node.js]: I guess the idea here is unpkg is the fallback to the fallback (GitHub) 😛

File: packages/gatsby-source-npm-package-search/src/gatsby-node.js (hunk 1 above)
Comment (wardpeet) [reply packages/gatsby-source-npm-package-search/src/gatsby-node.js]: Oh yeah my bad, I refactored it before pushing 🤦‍♂️

File: packages/gatsby-source-npm-package-search/src/gatsby-node.js (hunk 2 above)
Comment (wardpeet) [reply packages/gatsby-source-npm-package-search/src/gatsby-node.js]: Maybe I should skip github and go straight from unpkg. I copied the github code from yarn and noticed that we didn't get all Readme files as not all package.json have correct info for monorepos. 

Unpkg is always right so I'll probably drop github

**Comment by Haroenv** [comment]
nice way of handling with the fallbacks. I still want to get rid of this necessity at some point, but it seemed to be unfixable on npm's side

**Pull request merged**